src/main/java/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,26 @@
+package service.impl;
+
+import service.OperationStrategy;
+import service.strategy.BalanceHandler;
+import service.strategy.Handler;
+import service.strategy.PurchaseHandler;
+import service.strategy.ReturnHandler;
+import service.strategy.SupplyHandler;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    @Override
+    public Handler get(String[] str) {
```

Comment:
instead it's better to create and initilize`Map<String, Handler>` and fill it with the Handler implementations not to create new object of Handler per each call.